### PR TITLE
Prevent JavaScript error on malformed URI

### PR DESCRIPTION
**Pull request by HugoGiraudel**

### CODE REVIEW

**Comment by HugoGiraudel** [comment]
@Tom-Bonnike We do. Added.

File: src/helpers/getLocaleFromRequest/spec.js [hunk 1]
```diff
@@ -47,6 +47,13 @@ describe('The `getLocaleFromRequest` helper', function () {
     expect(actual).toEqual(expected)
   })
 
+  it('should not throw if failing to decode query from request', function () {
+    var request = getMockedRequest({ query: { locale: '/at//%c0%ae%c0%ae%c0%ae%c0%c0%a%c0%c0%c0%c0%af%c0%af%c0%af%c0%c0%ae%c0%c0%ae%c0%ae%c0%afwindowswin.ini' } })
```
Comment (Tom-Bonnike) [inline src/helpers/getLocaleFromRequest/spec.js]: 😄

File: src/helpers/getLocaleFromRequest/index.js [hunk 2]
```diff
@@ -17,7 +17,11 @@ function getQueryFromRequest (request) {
       return param
     }
 
-    return request.query[key] ? decodeURIComponent(request.query[key]) : null
+    try {
+      return request.query[key] ? decodeURIComponent(request.query[key]) : null
+    } catch (error) {
+      return null
+    }
```
Comment (Tom-Bonnike) [inline src/helpers/getLocaleFromRequest/index.js]: I prefered when you let it fail before but doesn’t matter 😄

Comment (HugoGiraudel) [reply src/helpers/getLocaleFromRequest/index.js]: Well but that’s the entire problem.

File: src/helpers/getLocaleFromRequest/spec.js (hunk 1 above)
Comment (HugoGiraudel) [reply src/helpers/getLocaleFromRequest/spec.js]: This is an example of URL used to DDoS us.

Comment (Tom-Bonnike) [reply src/helpers/getLocaleFromRequest/spec.js]: I know. :)

File: src/helpers/getLocaleFromRequest/index.js (hunk 2 above)
Comment (Tom-Bonnike) [reply src/helpers/getLocaleFromRequest/index.js]: No. What I meant is, before you rebased you wrote:

```javascript
try {
  return decodeURIComponent(request.query[key])
} catch (error) {
  return null
}
```

Comment (Tom-Bonnike) [reply src/helpers/getLocaleFromRequest/index.js]: Oh but that throw an error. Ok :)

Comment (HugoGiraudel) [reply src/helpers/getLocaleFromRequest/index.js]: That doesn’t do the same thing, all tests were failing. `decodeURIComponent(undefined)
` === `'undefined'`.

**Pull request merged**